.circleci/config.yml

```diff
@@ -529,6 +529,112 @@ jobs:
+      test:
+        description: 'the test that should be run multiple times'
+        type: string
+      runs:
+        description: 'number of times that the test should be run in total'
+        type: integer
+        default: 1600
+    docker:
+      - image: '<< parameters.image >>:<< parameters.image_tag >><< pipeline.parameters.image_suffix >>'
+    working_directory: /home/circleci/project
+    resource_class: small
+    steps:
+      - checkout
+      - attach_workspace:
+          at: .
+      - run:
+          name: 'Install Extension'
+          command: |
+            tar xfv "${CIRCLE_WORKING_DIRECTORY}/install-${PG_MAJOR}.tar" --directory /
+      - run:
+          name: 'Configure'
+          command: |
+            chown -R circleci .
+            gosu circleci ./configure --without-pg-version-check
+      - run:
+          name: 'Enable core dumps'
+          command: |
+            ulimit -c unlimited
+      - run:
+          name: 'Run minimal tests'
+          command: |
+            gosu circleci make -C src/test/regress << parameters.make >> EXTRA_TESTS="$(for i in $(seq << parameters.runs >> | circleci tests split); do echo -n '<< parameters.test >> ' ; done)"
+          no_output_timeout: 2m
+      - run:
+          name: 'Regressions'
+          command: |
+            if [ -f "src/test/regress/regression.diffs" ]; then
+              cat src/test/regress/regression.diffs
+              exit 1
+            fi
+          when: on_fail
+      - run:
+          name: 'Copy coredumps'
+          command: |
+            mkdir -p /tmp/core_dumps
+            if ls core.* 1> /dev/null 2>&1; then
+                cp core.* /tmp/core_dumps
+            fi
+          when: on_fail
+      - store_artifacts:
+          name: 'Save regressions'
+          path: src/test/regress/regression.diffs
+      - store_artifacts:
+          name: 'Save mitmproxy output (failure test specific)'
+          path: src/test/regress/proxy.output
+      - store_artifacts:
+          name: 'Save results'
+          path: src/test/regress/results/
+      - store_artifacts:
+          name: 'Save core dumps'
+          path: /tmp/core_dumps
+      - store_artifacts:
+          name: 'Save coordinator log'
+          path: src/test/regress/tmp_check/master/log
+      - store_artifacts:
+          name: 'Save worker1 log'
+          path: src/test/regress/tmp_check/worker.57637/log
+      - store_artifacts:
+          name: 'Save worker2 log'
+          path: src/test/regress/tmp_check/worker.57638/log
+
+
+# workflows:
+#   version: 2
+#   build_and_test:
+#     jobs:
+#       - build:
+#           name: build-13
+#           pg_major: 13
+#           image_tag: '<< pipeline.parameters.pg13_version >>'
+#
+#       - test-flakyness:
+#           name: 'test-13_flaky'
+#           pg_major: 13
+#           image_tag: '<< pipeline.parameters.pg13_version >>'
+#           requires: [build-13]
+#           test: adaptive_executor
```

Comment:
I would have expected this part as a separate workflow like below, with a condition on the workflow to not trigger on normal builds. Example: https://support.circleci.com/hc/en-us/articles/360050351292?input_string=how+can+i+share+the+data+between+all+the+jobs+in+a+workflow

Even though the blog only gives an example of manually triggering it with curl I think you can nowadays trigger it from the web interface with `trigger pipeline`. You can then pass extra parameters in to enable this workflow and with the name of the test to be run.

Reply:
Got it to work with the second workflow that's disabled by default: 
https://app.circleci.com/pipelines/github/citusdata/citus/26732/workflows/28825eda-4975-4b92-94aa-692c2cffe11c

This is definitely better than the commented out thing, but the circleci interface is not very userfriendly for this. So likely I'll still change the parameter default in the config myself when debugging a flaky test.

Reply:
wonder if a readme would help, but I think we can iterate on this based on experience while using.